packages/fxa-settings/src/components/PageWithLoggedInStatusState/index.tsx

```diff
@@ -0,0 +1,28 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React, { useState, useEffect } from 'react';
+import { useInitialState } from '../../models';
+import { RouteComponentProps } from '@reach/router';
+
+export const PageWithLoggedInStatusState = (
+  props: any & RouteComponentProps & { Page: React.ElementType }
+) => {
+  const { Page } = props;
+  const { loading, error } = useInitialState();
+
+  const [isSignedIn, setIsSignedIn] = useState<boolean>();
+
+  useEffect(() => {
+    if (!loading && error?.message.includes('Invalid token')) {
```

Comment:
Is there an errno we can check against instead, or another assertion against the error? This error message "Invalid token" is also present when a user has an expired reset PW link.

Reply:
I think there is actually, i think it's a 401. I'll update.

Reply:
Maybe it should be _both_? `gql.ts` sets `reauth=true` for both `error.message === 'Invalid token'` and `networkError.statusCode === 401`

Reply:
Ohhhh I can check for both then, if that feels like the right thing to do!